Exception in editing the user survey responses

### Describe the bug

While i have checked https://github.com/decidim/decidim/pull/16659 implementation, I have used the user form to reply to the questions. While i have not triggered any conditional field, i have been able to send my responses. 
When i have tried to edit my responses, i got the following error: 

### To Reproduce

1. Go to admin / participatory processes/ components / survey 
2. Configure a survey to allow users to edit their answers
3. Visit the web interface, Answer a survey, but to not trigger any conditional fields. 
4. Click [Edit your responses](http://localhost:3000/en/processes/defend-linger/f/10/surveys/1/edit) 
5. See error

### Expected behavior

To be able to edit my answers surveys without any errors. 

### Screenshots

<img width="1696" height="763" alt="Image" src="https://github.com/user-attachments/assets/87928361-9c2c-47a5-b351-bddb8c16e45c" />

### Stacktrace

```
16:45:46 web.1         | ActionView::Template::Error (undefined method 'responses' for nil)
16:45:46 web.1         | Caused by: NoMethodError (undefined method 'responses' for nil)
16:45:46 web.1         | 
16:45:46 web.1         | Information for: ActionView::Template::Error (undefined method 'responses' for nil):
16:45:46 web.1         |     1: <div>
16:45:46 web.1         |     2:   <%= response_form.text_field :body, label: false, id: field_id, disabled:, maxlength:, class: "w-full", required: false, data: { controller: "character-counter" } %>
16:45:46 web.1         |     3: </div>
16:45:46 web.1         |   
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/forms/decidim/forms/response_form.rb:70:in 'block in Decidim::Forms::ResponseForm#mandatory_conditions_fulfilled?'
16:45:46 web.1         | activerecord (8.1.3) lib/active_record/relation/delegation.rb:100:in 'Array#each'
16:45:46 web.1         | activerecord (8.1.3) lib/active_record/relation/delegation.rb:100:in 'ActiveRecord::Delegation#each'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/forms/decidim/forms/response_form.rb:69:in 'Enumerable#all?'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/forms/decidim/forms/response_form.rb:69:in 'Decidim::Forms::ResponseForm#mandatory_conditions_fulfilled?'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/forms/decidim/forms/response_form.rb:65:in 'Decidim::Forms::ResponseForm#display_conditions_fulfilled?'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/forms/decidim/forms/response_form.rb:99:in 'Decidim::Forms::ResponseForm#mandatory_body?'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-core/lib/decidim/form_builder.rb:564:in 'Decidim::FormBuilder#attribute_required?'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-core/lib/decidim/form_builder.rb:544:in 'Decidim::FormBuilder#extract_validations'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-core/lib/decidim/form_builder.rb:518:in 'Decidim::FormBuilder#field_with_validations'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-core/lib/decidim/form_builder.rb:496:in 'block in Decidim::FormBuilder#field'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-core/lib/decidim/form_builder.rb:622:in 'Decidim::FormBuilder#custom_label'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-core/lib/decidim/form_builder.rb:495:in 'Decidim::FormBuilder#field'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-core/lib/decidim/legacy_form_builder.rb:20:in 'block (2 levels) in <class:LegacyFormBuilder>'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/views/decidim/forms/questionnaires/responses/_short_response.html.erb:2
16:45:46 web.1         | actionview (8.1.3) lib/action_view/base.rb:281:in 'Kernel#public_send'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/base.rb:281:in 'ActionView::Base#_run'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/template.rb:284:in 'block in ActionView::Template#render'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:212:in 'ActiveSupport::Notifications.instrument'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/template.rb:589:in 'ActionView::Template#instrument_render_template'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/template.rb:272:in 'ActionView::Template#render'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/partial_renderer.rb:268:in 'block in ActionView::PartialRenderer#render_partial_template'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:210:in 'block in ActiveSupport::Notifications.instrument'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications/instrumenter.rb:58:in 'ActiveSupport::Notifications::Instrumenter#instrument'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:210:in 'ActiveSupport::Notifications.instrument'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/partial_renderer.rb:262:in 'ActionView::PartialRenderer#render_partial_template'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/partial_renderer.rb:253:in 'ActionView::PartialRenderer#render'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/renderer.rb:78:in 'ActionView::Renderer#render_partial_to_object'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/renderer.rb:29:in 'ActionView::Renderer#render_to_object'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/renderer.rb:24:in 'ActionView::Renderer#render'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/rendering_helper.rb:145:in 'block in ActionView::Helpers::RenderingHelper#render'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/base.rb:308:in 'ActionView::Base#in_rendering_context'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/rendering_helper.rb:141:in 'ActionView::Helpers::RenderingHelper#render'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/views/decidim/forms/questionnaires/_response.html.erb:56
16:45:46 web.1         | actionview (8.1.3) lib/action_view/base.rb:281:in 'Kernel#public_send'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/base.rb:281:in 'ActionView::Base#_run'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/template.rb:284:in 'block in ActionView::Template#render'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:212:in 'ActiveSupport::Notifications.instrument'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/template.rb:589:in 'ActionView::Template#instrument_render_template'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/template.rb:272:in 'ActionView::Template#render'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/partial_renderer.rb:268:in 'block in ActionView::PartialRenderer#render_partial_template'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:210:in 'block in ActiveSupport::Notifications.instrument'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications/instrumenter.rb:58:in 'ActiveSupport::Notifications::Instrumenter#instrument'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:210:in 'ActiveSupport::Notifications.instrument'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/partial_renderer.rb:262:in 'ActionView::PartialRenderer#render_partial_template'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/partial_renderer.rb:253:in 'ActionView::PartialRenderer#render'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/renderer.rb:78:in 'ActionView::Renderer#render_partial_to_object'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/renderer.rb:49:in 'ActionView::Renderer#render_partial'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/rendering_helper.rb:152:in 'ActionView::Helpers::RenderingHelper#render'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/views/decidim/forms/questionnaires/_questionnaire.html.erb:36
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/capture_helper.rb:50:in 'block in ActionView::Helpers::CaptureHelper#capture'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/buffers.rb:75:in 'ActionView::OutputBuffer#capture'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/capture_helper.rb:50:in 'ActionView::Helpers::CaptureHelper#capture'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/form_helper.rb:1088:in 'ActionView::Helpers::FormHelper#fields'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/form_helper.rb:1031:in 'ActionView::Helpers::FormHelper#fields_for'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/views/decidim/forms/questionnaires/_questionnaire.html.erb:35
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/views/decidim/forms/questionnaires/_questionnaire.html.erb:29:in 'Array#each'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/views/decidim/forms/questionnaires/_questionnaire.html.erb:29
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/views/decidim/forms/questionnaires/_questionnaire.html.erb:20:in 'Array#each'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/views/decidim/forms/questionnaires/_questionnaire.html.erb:20:in 'Enumerable#each_with_index'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/views/decidim/forms/questionnaires/_questionnaire.html.erb:20
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/capture_helper.rb:50:in 'block in ActionView::Helpers::CaptureHelper#capture'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/buffers.rb:75:in 'ActionView::OutputBuffer#capture'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/capture_helper.rb:50:in 'ActionView::Helpers::CaptureHelper#capture'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/form_helper.rb:775:in 'ActionView::Helpers::FormHelper#form_with'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/form_helper.rb:462:in 'ActionView::Helpers::FormHelper#form_for'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-core/app/helpers/decidim/decidim_form_helper.rb:30:in 'Decidim::DecidimFormHelper#decidim_form_for'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/views/decidim/forms/questionnaires/_questionnaire.html.erb:13
16:45:46 web.1         | actionview (8.1.3) lib/action_view/base.rb:281:in 'Kernel#public_send'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/base.rb:281:in 'ActionView::Base#_run'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/template.rb:284:in 'block in ActionView::Template#render'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:212:in 'ActiveSupport::Notifications.instrument'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/template.rb:589:in 'ActionView::Template#instrument_render_template'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/template.rb:272:in 'ActionView::Template#render'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/partial_renderer.rb:268:in 'block in ActionView::PartialRenderer#render_partial_template'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:210:in 'block in ActiveSupport::Notifications.instrument'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications/instrumenter.rb:58:in 'ActiveSupport::Notifications::Instrumenter#instrument'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:210:in 'ActiveSupport::Notifications.instrument'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/partial_renderer.rb:262:in 'ActionView::PartialRenderer#render_partial_template'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/partial_renderer.rb:253:in 'ActionView::PartialRenderer#render'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/renderer.rb:78:in 'ActionView::Renderer#render_partial_to_object'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/renderer.rb:29:in 'ActionView::Renderer#render_to_object'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/renderer.rb:24:in 'ActionView::Renderer#render'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/rendering_helper.rb:145:in 'block in ActionView::Helpers::RenderingHelper#render'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/base.rb:308:in 'ActionView::Base#in_rendering_context'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/rendering_helper.rb:141:in 'ActionView::Helpers::RenderingHelper#render'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/views/decidim/forms/questionnaires/edit.html.erb:37
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/capture_helper.rb:50:in 'block in ActionView::Helpers::CaptureHelper#capture'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/buffers.rb:75:in 'ActionView::OutputBuffer#capture'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/capture_helper.rb:50:in 'ActionView::Helpers::CaptureHelper#capture'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/rendering_helper.rb:211:in 'ActionView::Helpers::RenderingHelper#_layout_for'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/partial_renderer.rb:269:in 'block (2 levels) in ActionView::PartialRenderer#render_partial_template'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-core/app/views/layouts/decidim/shared/_layout_center.html.erb:14
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/capture_helper.rb:50:in 'block in ActionView::Helpers::CaptureHelper#capture'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/buffers.rb:75:in 'ActionView::OutputBuffer#capture'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/capture_helper.rb:50:in 'ActionView::Helpers::CaptureHelper#capture'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/tag_helper.rb:521:in 'ActionView::Helpers::TagHelper#content_tag'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-core/app/views/layouts/decidim/shared/_layout_center.html.erb:13
16:45:46 web.1         | actionview (8.1.3) lib/action_view/base.rb:281:in 'Kernel#public_send'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/base.rb:281:in 'ActionView::Base#_run'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/template.rb:284:in 'block in ActionView::Template#render'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:212:in 'ActiveSupport::Notifications.instrument'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/template.rb:589:in 'ActionView::Template#instrument_render_template'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/template.rb:272:in 'ActionView::Template#render'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/partial_renderer.rb:268:in 'block in ActionView::PartialRenderer#render_partial_template'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:210:in 'block in ActiveSupport::Notifications.instrument'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications/instrumenter.rb:58:in 'ActiveSupport::Notifications::Instrumenter#instrument'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:210:in 'ActiveSupport::Notifications.instrument'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/partial_renderer.rb:262:in 'ActionView::PartialRenderer#render_partial_template'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/partial_renderer.rb:253:in 'ActionView::PartialRenderer#render'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/renderer.rb:78:in 'ActionView::Renderer#render_partial_to_object'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/renderer.rb:49:in 'ActionView::Renderer#render_partial'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/rendering_helper.rb:143:in 'block in ActionView::Helpers::RenderingHelper#render'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/base.rb:308:in 'ActionView::Base#in_rendering_context'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/helpers/rendering_helper.rb:141:in 'ActionView::Helpers::RenderingHelper#render'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-forms/app/views/decidim/forms/questionnaires/edit.html.erb:9
16:45:46 web.1         | actionview (8.1.3) lib/action_view/base.rb:281:in 'Kernel#public_send'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/base.rb:281:in 'ActionView::Base#_run'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/template.rb:284:in 'block in ActionView::Template#render'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:212:in 'ActiveSupport::Notifications.instrument'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/template.rb:589:in 'ActionView::Template#instrument_render_template'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/template.rb:272:in 'ActionView::Template#render'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/template_renderer.rb:66:in 'block (2 levels) in ActionView::TemplateRenderer#render_template'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:210:in 'block in ActiveSupport::Notifications.instrument'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications/instrumenter.rb:58:in 'ActiveSupport::Notifications::Instrumenter#instrument'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:210:in 'ActiveSupport::Notifications.instrument'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/template_renderer.rb:60:in 'block in ActionView::TemplateRenderer#render_template'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/template_renderer.rb:76:in 'block in ActionView::TemplateRenderer#render_with_layout'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:210:in 'block in ActiveSupport::Notifications.instrument'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications/instrumenter.rb:58:in 'ActiveSupport::Notifications::Instrumenter#instrument'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:210:in 'ActiveSupport::Notifications.instrument'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/template_renderer.rb:75:in 'ActionView::TemplateRenderer#render_with_layout'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/template_renderer.rb:59:in 'ActionView::TemplateRenderer#render_template'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/template_renderer.rb:11:in 'ActionView::TemplateRenderer#render'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/renderer.rb:58:in 'ActionView::Renderer#render_template_to_object'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/renderer/renderer.rb:31:in 'ActionView::Renderer#render_to_object'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/rendering.rb:136:in 'block in ActionView::Rendering#_render_template'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/base.rb:308:in 'ActionView::Base#in_rendering_context'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/rendering.rb:135:in 'ActionView::Rendering#_render_template'
16:45:46 web.1         | actionpack (8.1.3) lib/action_controller/metal/streaming.rb:179:in 'ActionController::Streaming#_render_template'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/rendering.rb:122:in 'ActionView::Rendering#render_to_body'
16:45:46 web.1         | actionpack (8.1.3) lib/action_controller/metal/rendering.rb:192:in 'ActionController::Rendering#render_to_body'
16:45:46 web.1         | actionpack (8.1.3) lib/action_controller/metal/renderers.rb:155:in 'ActionController::Renderers#render_to_body'
16:45:46 web.1         | actionpack (8.1.3) lib/abstract_controller/rendering.rb:28:in 'AbstractController::Rendering#render'
16:45:46 web.1         | actionpack (8.1.3) lib/action_controller/metal/rendering.rb:173:in 'ActionController::Rendering#render'
16:45:46 web.1         | actionpack (8.1.3) lib/action_controller/metal/instrumentation.rb:31:in 'block (2 levels) in ActionController::Instrumentation#render'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/benchmark.rb:17:in 'ActiveSupport::Benchmark.realtime'
16:45:46 web.1         | actionpack (8.1.3) lib/action_controller/metal/instrumentation.rb:31:in 'block in ActionController::Instrumentation#render'
16:45:46 web.1         | actionpack (8.1.3) lib/action_controller/metal/instrumentation.rb:100:in 'ActionController::Instrumentation#cleanup_view_runtime'
16:45:46 web.1         | activerecord (8.1.3) lib/active_record/railties/controller_runtime.rb:48:in 'ActiveRecord::Railties::ControllerRuntime#cleanup_view_runtime'
16:45:46 web.1         | actionpack (8.1.3) lib/action_controller/metal/instrumentation.rb:30:in 'ActionController::Instrumentation#render'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-surveys/app/controllers/decidim/surveys/surveys_controller.rb:29:in 'Decidim::Surveys::SurveysController#edit'
16:45:46 web.1         | actionpack (8.1.3) lib/action_controller/metal/basic_implicit_render.rb:8:in 'ActionController::BasicImplicitRender#send_action'
16:45:46 web.1         | actionpack (8.1.3) lib/abstract_controller/base.rb:221:in 'AbstractController::Base#process_action'
16:45:46 web.1         | actionpack (8.1.3) lib/action_controller/metal/rendering.rb:199:in 'ActionController::Rendering#process_action'
16:45:46 web.1         | actionpack (8.1.3) lib/abstract_controller/callbacks.rb:267:in 'block in AbstractController::Callbacks#process_action'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/callbacks.rb:121:in 'block in ActiveSupport::Callbacks#run_callbacks'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/core_ext/time/zones.rb:65:in 'Time.use_zone'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-core/app/controllers/concerns/decidim/use_organization_time_zone.rb:21:in 'Decidim::ApplicationController#use_organization_time_zone'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/callbacks.rb:130:in 'block in ActiveSupport::Callbacks#run_callbacks'
16:45:46 web.1         | i18n (1.14.8) lib/i18n.rb:354:in 'I18n::Base#with_locale'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-core/app/controllers/concerns/decidim/locale_switcher.rb:25:in 'Decidim::ApplicationController#switch_locale'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/callbacks.rb:130:in 'block in ActiveSupport::Callbacks#run_callbacks'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/callbacks.rb:141:in 'ActiveSupport::Callbacks#run_callbacks'
16:45:46 web.1         | actionpack (8.1.3) lib/abstract_controller/callbacks.rb:266:in 'AbstractController::Callbacks#process_action'
16:45:46 web.1         | actionpack (8.1.3) lib/action_controller/metal/rescue.rb:36:in 'ActionController::Rescue#process_action'
16:45:46 web.1         | actionpack (8.1.3) lib/action_controller/metal/instrumentation.rb:76:in 'block in ActionController::Instrumentation#process_action'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:210:in 'block in ActiveSupport::Notifications.instrument'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications/instrumenter.rb:58:in 'ActiveSupport::Notifications::Instrumenter#instrument'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/notifications.rb:210:in 'ActiveSupport::Notifications.instrument'
16:45:46 web.1         | actionpack (8.1.3) lib/action_controller/metal/instrumentation.rb:75:in 'ActionController::Instrumentation#process_action'
16:45:46 web.1         | actionpack (8.1.3) lib/action_controller/metal/params_wrapper.rb:259:in 'ActionController::ParamsWrapper#process_action'
16:45:46 web.1         | activerecord (8.1.3) lib/active_record/railties/controller_runtime.rb:39:in 'ActiveRecord::Railties::ControllerRuntime#process_action'
16:45:46 web.1         | actionpack (8.1.3) lib/abstract_controller/base.rb:154:in 'AbstractController::Base#process'
16:45:46 web.1         | actionview (8.1.3) lib/action_view/rendering.rb:40:in 'ActionView::Rendering#process'
16:45:46 web.1         | actionpack (8.1.3) lib/action_controller/metal.rb:252:in 'ActionController::Metal#dispatch'
16:45:46 web.1         | actionpack (8.1.3) lib/action_controller/metal.rb:335:in 'ActionController::Metal.dispatch'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/routing/route_set.rb:65:in 'ActionDispatch::Routing::RouteSet::Dispatcher#dispatch'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/routing/route_set.rb:50:in 'ActionDispatch::Routing::RouteSet::Dispatcher#serve'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:35:in 'block in ActionDispatch::Journey::Router#serve'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:86:in 'block in ActionDispatch::Journey::Router#recognize'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:66:in 'Array#each'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:66:in 'ActionDispatch::Journey::Router#recognize'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:31:in 'ActionDispatch::Journey::Router#serve'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/routing/route_set.rb:906:in 'ActionDispatch::Routing::RouteSet#call'
16:45:46 web.1         | railties (8.1.3) lib/rails/engine.rb:534:in 'Rails::Engine#call'
16:45:46 web.1         | railties (8.1.3) lib/rails/railtie.rb:225:in 'Kernel#public_send'
16:45:46 web.1         | railties (8.1.3) lib/rails/railtie.rb:225:in 'Rails::Railtie.method_missing'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/routing/mapper.rb:33:in 'block in <class:Constraints>'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/routing/mapper.rb:62:in 'ActionDispatch::Routing::Mapper::Constraints#serve'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:35:in 'block in ActionDispatch::Journey::Router#serve'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:92:in 'block in ActionDispatch::Journey::Router#recognize'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:66:in 'Array#each'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:66:in 'ActionDispatch::Journey::Router#recognize'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:31:in 'ActionDispatch::Journey::Router#serve'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/routing/route_set.rb:906:in 'ActionDispatch::Routing::RouteSet#call'
16:45:46 web.1         | railties (8.1.3) lib/rails/engine.rb:534:in 'Rails::Engine#call'
16:45:46 web.1         | railties (8.1.3) lib/rails/railtie.rb:225:in 'Kernel#public_send'
16:45:46 web.1         | railties (8.1.3) lib/rails/railtie.rb:225:in 'Rails::Railtie.method_missing'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/routing/mapper.rb:33:in 'block in <class:Constraints>'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/routing/mapper.rb:62:in 'ActionDispatch::Routing::Mapper::Constraints#serve'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:35:in 'block in ActionDispatch::Journey::Router#serve'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:92:in 'block in ActionDispatch::Journey::Router#recognize'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:66:in 'Array#each'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:66:in 'ActionDispatch::Journey::Router#recognize'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:31:in 'ActionDispatch::Journey::Router#serve'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/routing/route_set.rb:906:in 'ActionDispatch::Routing::RouteSet#call'
16:45:46 web.1         | railties (8.1.3) lib/rails/engine/lazy_route_set.rb:60:in 'Rails::Engine::LazyRouteSet#call'
16:45:46 web.1         | railties (8.1.3) lib/rails/engine.rb:534:in 'Rails::Engine#call'
16:45:46 web.1         | railties (8.1.3) lib/rails/railtie.rb:225:in 'Kernel#public_send'
16:45:46 web.1         | railties (8.1.3) lib/rails/railtie.rb:225:in 'Rails::Railtie.method_missing'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/routing/mapper.rb:33:in 'block in <class:Constraints>'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/routing/mapper.rb:62:in 'ActionDispatch::Routing::Mapper::Constraints#serve'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:35:in 'block in ActionDispatch::Journey::Router#serve'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:92:in 'block in ActionDispatch::Journey::Router#recognize'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:66:in 'Array#each'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:66:in 'ActionDispatch::Journey::Router#recognize'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/journey/router.rb:31:in 'ActionDispatch::Journey::Router#serve'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/routing/route_set.rb:906:in 'ActionDispatch::Routing::RouteSet#call'
16:45:46 web.1         | railties (8.1.3) lib/rails/engine/lazy_route_set.rb:60:in 'Rails::Engine::LazyRouteSet#call'
16:45:46 web.1         | batch-loader (2.0.6) lib/batch_loader/middleware.rb:11:in 'BatchLoader::Middleware#call'
16:45:46 web.1         | bullet (8.1.1) lib/bullet/rack.rb:21:in 'Bullet::Rack#call'
16:45:46 web.1         | omniauth (2.1.4) lib/omniauth/strategy.rb:202:in 'OmniAuth::Strategy#call!'
16:45:46 web.1         | omniauth (2.1.4) lib/omniauth/strategy.rb:169:in 'OmniAuth::Strategy#call'
16:45:46 web.1         | omniauth (2.1.4) lib/omniauth/strategy.rb:202:in 'OmniAuth::Strategy#call!'
16:45:46 web.1         | omniauth (2.1.4) lib/omniauth/strategy.rb:169:in 'OmniAuth::Strategy#call'
16:45:46 web.1         | omniauth (2.1.4) lib/omniauth/strategy.rb:202:in 'OmniAuth::Strategy#call!'
16:45:46 web.1         | omniauth (2.1.4) lib/omniauth/strategy.rb:169:in 'OmniAuth::Strategy#call'
16:45:46 web.1         | omniauth (2.1.4) lib/omniauth/strategy.rb:202:in 'OmniAuth::Strategy#call!'
16:45:46 web.1         | omniauth (2.1.4) lib/omniauth/strategy.rb:169:in 'OmniAuth::Strategy#call'
16:45:46 web.1         | omniauth (2.1.4) lib/omniauth/builder.rb:48:in 'OmniAuth::Builder#call'
16:45:46 web.1         | warden-jwt_auth (0.12.0) lib/warden/jwt_auth/middleware/token_dispatcher.rb:20:in 'Warden::JWTAuth::Middleware::TokenDispatcher#call'
16:45:46 web.1         | warden-jwt_auth (0.12.0) lib/warden/jwt_auth/middleware/revocation_manager.rb:21:in 'Warden::JWTAuth::Middleware::RevocationManager#call'
16:45:46 web.1         | rack (3.2.6) lib/rack/builder.rb:283:in 'Rack::Builder#call'
16:45:46 web.1         | warden-jwt_auth (0.12.0) lib/warden/jwt_auth/middleware.rb:22:in 'Warden::JWTAuth::Middleware#call'
16:45:46 web.1         | warden (1.2.9) lib/warden/manager.rb:36:in 'block in Warden::Manager#call'
16:45:46 web.1         | warden (1.2.9) lib/warden/manager.rb:34:in 'Kernel#catch'
16:45:46 web.1         | warden (1.2.9) lib/warden/manager.rb:34:in 'Warden::Manager#call'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-core/lib/decidim/middleware/strip_x_forwarded_host.rb:12:in 'Decidim::Middleware::StripXForwardedHost#call'
16:45:46 web.1         | /home/alecslupu/Sites/decidim/develop/decidim-core/lib/decidim/middleware/current_organization.rb:22:in 'Decidim::Middleware::CurrentOrganization#call'
16:45:46 web.1         | rack (3.2.6) lib/rack/tempfile_reaper.rb:20:in 'Rack::TempfileReaper#call'
16:45:46 web.1         | rack (3.2.6) lib/rack/etag.rb:29:in 'Rack::ETag#call'
16:45:46 web.1         | rack (3.2.6) lib/rack/conditional_get.rb:31:in 'Rack::ConditionalGet#call'
16:45:46 web.1         | rack (3.2.6) lib/rack/head.rb:15:in 'Rack::Head#call'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/http/content_security_policy.rb:38:in 'ActionDispatch::ContentSecurityPolicy::Middleware#call'
16:45:46 web.1         | rack-session (2.1.2) lib/rack/session/abstract/id.rb:274:in 'Rack::Session::Abstract::Persisted#context'
16:45:46 web.1         | rack-session (2.1.2) lib/rack/session/abstract/id.rb:268:in 'Rack::Session::Abstract::Persisted#call'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/middleware/cookies.rb:708:in 'ActionDispatch::Cookies#call'
16:45:46 web.1         | activerecord (8.1.3) lib/active_record/migration.rb:671:in 'ActiveRecord::Migration::CheckPending#call'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/middleware/callbacks.rb:31:in 'block in ActionDispatch::Callbacks#call'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/callbacks.rb:101:in 'ActiveSupport::Callbacks#run_callbacks'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/middleware/callbacks.rb:30:in 'ActionDispatch::Callbacks#call'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/middleware/executor.rb:20:in 'ActionDispatch::Executor#call'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/middleware/actionable_exceptions.rb:18:in 'ActionDispatch::ActionableExceptions#call'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/middleware/debug_exceptions.rb:31:in 'ActionDispatch::DebugExceptions#call'
16:45:46 web.1         | web-console (4.3.0) lib/web_console/middleware.rb:132:in 'WebConsole::Middleware#call_app'
16:45:46 web.1         | web-console (4.3.0) lib/web_console/middleware.rb:28:in 'block in WebConsole::Middleware#call'
16:45:46 web.1         | web-console (4.3.0) lib/web_console/middleware.rb:17:in 'Kernel#catch'
16:45:46 web.1         | web-console (4.3.0) lib/web_console/middleware.rb:17:in 'WebConsole::Middleware#call'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/middleware/show_exceptions.rb:32:in 'ActionDispatch::ShowExceptions#call'
16:45:46 web.1         | railties (8.1.3) lib/rails/rack/logger.rb:41:in 'Rails::Rack::Logger#call_app'
16:45:46 web.1         | railties (8.1.3) lib/rails/rack/logger.rb:29:in 'Rails::Rack::Logger#call'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/middleware/remote_ip.rb:98:in 'ActionDispatch::RemoteIp#call'
16:45:46 web.1         | request_store (1.7.0) lib/request_store/middleware.rb:19:in 'RequestStore::Middleware#call'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/middleware/request_id.rb:34:in 'ActionDispatch::RequestId#call'
16:45:46 web.1         | rack (3.2.6) lib/rack/method_override.rb:28:in 'Rack::MethodOverride#call'
16:45:46 web.1         | activesupport (8.1.3) lib/active_support/cache/strategy/local_cache_middleware.rb:30:in 'ActiveSupport::Cache::Strategy::LocalCache::Middleware#call'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/middleware/server_timing.rb:61:in 'block in ActionDispatch::ServerTiming#call'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/middleware/server_timing.rb:26:in 'ActionDispatch::ServerTiming::Subscriber#collect_events'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/middleware/server_timing.rb:60:in 'ActionDispatch::ServerTiming#call'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/middleware/executor.rb:20:in 'ActionDispatch::Executor#call'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/middleware/static.rb:27:in 'ActionDispatch::Static#call'
16:45:46 web.1         | rack (3.2.6) lib/rack/sendfile.rb:131:in 'Rack::Sendfile#call'
16:45:46 web.1         | actionpack (8.1.3) lib/action_dispatch/middleware/host_authorization.rb:143:in 'ActionDispatch::HostAuthorization#call'
16:45:46 web.1         | rack-mini-profiler (4.0.1) lib/mini_profiler.rb:334:in 'Rack::MiniProfiler#call'
16:45:46 web.1         | shakapacker (9.7.0) lib/shakapacker/dev_server_proxy.rb:25:in 'Shakapacker::DevServerProxy#perform_request'
16:45:46 web.1         | rack-proxy (0.7.7) lib/rack/proxy.rb:87:in 'Rack::Proxy#call'
16:45:46 web.1         | rack-cors (3.0.0) lib/rack/cors.rb:102:in 'Rack::Cors#call'
16:45:46 web.1         | railties (8.1.3) lib/rails/engine.rb:534:in 'Rails::Engine#call'
16:45:46 web.1         | puma (7.2.0) lib/puma/configuration.rb:296:in 'Puma::Configuration::ConfigMiddleware#call'
16:45:46 web.1         | puma (7.2.0) lib/puma/request.rb:103:in 'block in Puma::Request#handle_request'
16:45:46 web.1         | puma (7.2.0) lib/puma/thread_pool.rb:355:in 'Puma::ThreadPool#with_force_shutdown'
16:45:46 web.1         | puma (7.2.0) lib/puma/request.rb:102:in 'Puma::Request#handle_request'
16:45:46 web.1         | puma (7.2.0) lib/puma/server.rb:503:in 'Puma::Server#process_client'
16:45:46 web.1         | puma (7.2.0) lib/puma/server.rb:262:in 'block in Puma::Server#run'
16:45:46 web.1         | puma (7.2.0) lib/puma/thread_pool.rb:182:in 'block in Puma::ThreadPool#spawn_thread'

```

### Extra data

- Device:
- Device OS:
- Browser:
- Decidim Version: develop
- Decidim installation:


### Additional context

_No response_

--- a/decidim-forms/app/forms/decidim/forms/questionnaire_form.rb
+++ b/decidim-forms/app/forms/decidim/forms/questionnaire_form.rb
@@ -30,7 +30,7 @@ module Decidim
 
       def add_responses!(questionnaire:, session_token:, ip_hash:)
         self.responses = questionnaire.questions.map do |question|
-          ResponseForm.from_model(Decidim::Forms::Response.where(question:, user: current_user, session_token:, ip_hash:).first_or_initialize)
+          ResponseForm.from_model(Decidim::Forms::Response.where(question:, user: current_user, session_token:, ip_hash:).first_or_initialize).with_context(context)
         end
       end
 

--- a/decidim-surveys/spec/system/survey_spec.rb
+++ b/decidim-surveys/spec/system/survey_spec.rb
@@ -198,6 +198,42 @@ describe "Respond a survey" do
         it_behaves_like "has embedded video in description", :question_description
       end
     end
+
+    context "when the survey allows to edit responses, and question has display conditions" do
+      let!(:question_two) do
+        create(:questionnaire_question,
+               questionnaire:,
+               question_type: "short_response",
+               position: 1,
+               options: [])
+      end
+      let!(:display_condition) do
+        create(:display_condition,
+               condition_type: "not_responded",
+               question: question_two,
+               condition_question: question)
+      end
+
+      before do
+        survey.update!(
+          allow_responses: true,
+          allow_editing_responses: true
+        )
+        login_as user, scope: :user
+        visit_component
+        click_on translated_attribute(questionnaire.title)
+        fill_in "questionnaire_responses_0", with: "test"
+        check "questionnaire_tos_agreement"
+        click_on "Submit"
+        expect(page).to have_callout("Survey successfully responded.")
+      end
+
+      it "allows to edit the responses" do
+        click_on "Edit your responses"
+        expect(page).to have_content(translated_attribute(survey.title))
+        expect(page).to have_content(translated_attribute(question.body))
+      end
+    end
   end
 
   context "when survey has a custom announcement" do

